qemu/unicorn_common.h

```diff
@@ -128,4 +128,13 @@ static inline void uc_common_init(struct uc_struct* uc)
     if (!uc->release)
         uc->release = release_common;
 }
+
+#define CHECK_REG_TYPE(type) do {             \
+    if (unlikely(*size < sizeof(type))) {     \
+        return UC_ERR_NOMEM;                  \
```

Comment:
This is not correct, I suggest `UC_ERR_ARG`

Reply:
Hmm, I wanted it to be distinct from "register not found" (which is UC_ERR_ARG) because it may hint the caller to try a different buffer size. I realize NOMEM is not quite accurate. Any other suggestions?

Reply:
I suggest a new error type, what's your thoughts? Maybe `UC_ERR_REG`. Note to add a new error, you also need to update `uc_strerror`.

Reply:
Changed to a new error `UC_ERR_OVERFLOW`.